controller.go

```diff
@@ -119,7 +120,15 @@ type driverData struct {
 	capability driverapi.Capability
 }
 
+type ipamData struct {
+	config    ipamapi.Config
+	allocator ipamapi.Allocator
+	// default address spaces are provided by ipam driver at registration time
+	defaultLocalAddressSpace, defaultGlobalAddressSpace string
```

Comment:
> The default address space is needed in case UI does not passes it (which will be always the case for first releases). We want the ipam driver to let us know which is the default addr space string.

got it, that was my understanding too.

> We need two default address spaces, one for local scoped networks (like for ex. bridge) and one to be used for global scoped networks (like for ex. overlay). Because on a docker host both kind of networks can co-exists.

I agree that libnetwork need to support both kind of networks, but does libnetwork expects the same IPAM driver to always work for local as well as global scoped networks? I ask as I don't think my driver would be able to serve both local and global scoped networks and it seems better to error out early that routing the request to driver. Would it be better to negotiate this as driver capability instead (like for NetworkDriver) and error out early if user tries to do this?